artemis-server/src/main/java/org/apache/activemq/artemis/core/server/cluster/quorum/QuorumManager.java

```diff
@@ -348,19 +377,42 @@ public synchronized void voteComplete() {
       }
    }
 
-   private Vote sendQuorumVote(ClusterControl clusterControl, SimpleString handler, Vote vote) {
+   private Vote sendQuorumVote(ClusterControl clusterControl, SimpleString handler, SimpleString oldHandlerName, Vote vote) {
       try {
          final ClientSessionFactoryInternal sessionFactory = clusterControl.getSessionFactory();
          final String remoteAddress = sessionFactory.getConnection().getRemoteAddress();
          ActiveMQServerLogger.LOGGER.sendingQuorumVoteRequest(remoteAddress, vote.toString());
-         QuorumVoteReplyMessage replyMessage = (QuorumVoteReplyMessage) clusterControl.getClusterChannel().get()
-            .sendBlocking(new QuorumVoteMessage(handler, vote), PacketImpl.QUORUM_VOTE_REPLY);
+
+         QuorumVoteReplyMessage replyMessage = null;
+
+         Channel clusterChannel = clusterControl.getClusterChannel().get();
+
+
+         // We first try the current packet with a medium timeout
+         replyMessage = (QuorumVoteReplyMessage) clusterChannel.sendBlocking(new QuorumVoteMessage(handler, vote), -1, PacketImpl.QUORUM_VOTE_REPLY, VOTE_RESPONSE_TIMEOUT, false);
+         logger.trace("This is the reply message from the current version = {}", replyMessage);
+
+         // if no response, we try the previous versions, with still a medium timeout
+         if (replyMessage == null && oldHandlerName != null) {
+            replyMessage = (QuorumVoteReplyMessage) clusterChannel.sendBlocking(new QuorumVoteMessage(oldHandlerName, vote), -1, PacketImpl.QUORUM_VOTE_REPLY, VOTE_RESPONSE_TIMEOUT, false);
+            logger.trace("This is the reply message from the older version = {}", replyMessage);
+         }
```

Comment:
What happens if a reply actually does arrive, but only after these short timeouts (2sec vs usual 30sec)? Ignored?

Also, in future could it check for a newer wire version and avoid doing this for newer servers?

Reply:
We could stop looking for this on a newer wire version... for sure.

I actually thought about upgrading the wire now, and skip these short timeouts. I had it written at some point and I reverted it... 

(I thought these retries would been enough)


To have these reply Messages being delayed more than 2 seconds here it would mean things are really bad for the cluster anyway. a Retry with the same serverID seemed enough to me here.


I could upgrade the wire now and not have these checks on current versions. WDYT?

Reply:
I will bump the release to avoid these checks if dealing with a current release